crypto/rsa/rsa_pk1.c

```diff
@@ -221,11 +221,6 @@ int RSA_padding_check_PKCS1_type_2(unsigned char *to, int tlen,
     msg_index = zero_index + 1;
     mlen = num - msg_index;
 
-    /*
-     * For good measure, do this check in constant time as well.
-     */
-    good &= constant_time_ge(tlen, mlen);
```

Comment:
If we could change that to
`good &= onstant_time_eq(mlen, expected_plaintext_length);`

That would make a __BIG__ (1/256) difference in effectiveness of a Bleichenbacher attack.

Reply:
but that has to be done at the application level, in constant time.
Additionally the application has to check at least a few known bytes in the payload,
for instance 0x03 0x03 for the SSL-version, but to be safe that should be 8 known bytes
in the plain text, they all need to be checked in constant time of course.

Reply:
Yes, I was also thinking about something like that, but we currently don't have that. We have at least talked about the need for a new API.

Reply:
What the comment is saying, is if we know how many bytes of zero-padding at the start of the _plaintext_ to expect,
that could be checked here and would effectively defeat the Bleichenbacher attack.